internal/internal_workflow_client.go

```diff
@@ -1057,6 +1057,48 @@ func (wc *WorkflowClient) GetWorkerTaskReachability(ctx context.Context, options
 	return converted, nil
 }
 
+// GetWorkflowExecutionOptions returns the current WorkflowExecutionOptions of an existing workflow.
+// NOTE: Experimental
+func (wc *WorkflowClient) GetWorkflowExecutionOptions(ctx context.Context, request GetWorkflowExecutionOptionsRequest) (WorkflowExecutionOptions, error) {
+	// no side-effects, i.e., GET, when UpdateMask is empty and WorkflowExecutionOptions is not nil
```

Comment:
Eek, I didn't notice in https://github.com/temporalio/api/pull/466 that we didn't provide a getter and we were expecting people to use an update call with an empty mask as a getter. I think we need an explicit getter and we need to fail on empty update mask server side. We need to clearly mark mutable RPC vs immutable where we can (it can also affect action recording logic) and it'll be silly for someone to have to `POST` to get options when using the HTTP API.

@carlydf - can we add a getter, or even better, can we just provide this information on describe response?

Reply:
All these are good reasons, in particular the HTTP API, @carlydf are we late to add the getter?

Reply:
I discussed with @carlydf  and @ShahabT . `DescribeWorkflowExecution` will return that info and it is not more expensive than returning the options in a getter. So I will remove the getter from the sdk and add a comment in Update to use `DescribeWorkflowExecution` as a getter... Is that ok @cretz ?

Reply:
> So I will remove the getter from the sdk and add a comment in Update to use DescribeWorkflowExecution as a getter

Great! I think this is the best of all worlds.